olive/passes/pytorch/pipeline_parallel.py

```diff
@@ -0,0 +1,153 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+# Automatically distribute PyTorch model using Pipeline Parallelism
+# --------------------------------------------------------------------------
+
+import logging
+import multiprocessing
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Callable, Dict
+
+import torch
+
+from olive.common.pydantic_v1 import validator
+from olive.hardware.accelerator import AcceleratorSpec, Device
+from olive.model import DistributedPyTorchModelHandler, PyTorchModelHandler
+from olive.passes import Pass
+from olive.passes.olive_pass import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class PipelineParallel:
+    def __init__(self, rank: int, world_size: int):
+        self.rank = rank
+        self.world_size = world_size
+
+    @abstractmethod
+    def replace_layers(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def restore_layers(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def split_layers(self, model: torch.nn.Module):
+        raise NotImplementedError
+
+
+class PyTorchPipelineParallel(Pass):
```

Comment:
it seems almost the logic is duplicated with that one in tensor_paralle.py except:
1. world_size input
2. generate PyTorchPipelineParallel or PyTorchTensorParallel
3. load rank weight or not.

Is it possible to make the redundancy code reusable?